src/resource/solidDataset.ts

```diff
@@ -871,14 +872,31 @@ export function getContainedResourceUrlAll(
 }
 
 /**
- * Given a {@link SolidDataset} representing a Container (see {@link isContainer}), verify that
- * all its containùent claims are valid. Containment of a resource is invalid if it doesn't
- * respect slash semantics, see https://solidproject.org/TR/protocol#resource-containment for
- * more details.
+ * Given a {@link SolidDataset} representing a [Container](https://solidproject.org/TR/protocol#resource-containment)
+ * (see {@link isContainer}), verify that all its containment claims are valid.
+ *
+ * Containment of a resource is valid if it respects
+ * [slash semantics](https://solidproject.org/TR/protocol#uri-slash-semantics). For the
+ * container at https://example.org/container/, the following resources are valid:
+ *  - https://example.org/container/resource
+ *  - https://example.org/container/subcontainer/
+ *
+ * The following resources are invalid:
+ *  - https://example.org/container/resource/invalid (not a direct child resource)
+ *  - https://example.org/container2 (not a child resource)
+ *  - https://domain2.example.org/container/resource (not a direct child resource)
  *
  * Resources for which containment is invalid are not included in the result set returned by
  * {@link getContainedResourceUrlAll}.
  *
+ * The ESS will always include containment triples that respect
+ * [slash semantics](https://solidproject.org/TR/protocol#uri-slash-semantics); and so
+ * this function will always return true for containers fetched from the ESS.
+ *
+ * It is recommended that this function *always* be used before reading the containment triples
+ * from a Pod using {@link getContainedResourceUrlAll} in order to detect unexpected behaviour,
+ * including malicious containment triples, when interacting with other Solid servers.
```

Comment:
I'd flip this sentence to lead with When interacting with other Solid servers, ...

Reply:
Ah - this isn't quite what I meant; I'm saying *always* use it (unconditionally), is the following clearer?
```
 * It is recommended that this function *always* be used before reading the containment triples
 * from a Pod using {@link getContainedResourceUrlAll}. This will allow you do detect unexpected behaviour from servers,
 * including malicious containment triples that could appear when interacting with other Solid servers.
 ```

Reply:
Ah ... if that's the case, I'd move up that up a bit so that we have 

> Given a SolidDataset ..., verify that all its containment claims are valid. Containment claims are valid if it respects slash semantics. 
> 
> For example, given container at https://example.org/container/:
> 
> - The following resources are valid:
> 
> - The following resources are invalid:
> 
> If a component claim is invalid, validateContainedResourceAll returns the invalid component's URL as part of its return object.
> 
> Note: It is recommended that this function always be used before calling
> {@link getContainedResourceUrlAll} since {@link getContainedResourceUrlAll} does not
> return Resources for which containment is invalid. Using the function in conjunction with {@link getContainedResourceUrlAll} allows for the detection of unexpected behaviour from servers,
> including malicious containment triples that could appear. Because ESS conforms to the Solid Protocol,i.e., respects slash semantics for its containment triples, `validateContainedResourceAll` returns true for containers fetched from ESS.


This way, we state

- What this function does, including explanation of what's invalid, and what it does return
- Recommendation: Use it with getContainedResourceUrlAll and why
- and the ESS statement becomes more of a throwaway.